### PR TITLE
Handle non-linearized MemberList in v3.4 for robustness tests

### PR DESCRIPTION
**Pull request by serathius**
cc @ahrtr @jmhbnz 

Fixes https://github.com/etcd-io/etcd/issues/18929

### CODE REVIEW

File: tests/robustness/failpoint/cluster.go [hunk 1]
```diff
@@ -137,11 +138,21 @@ func (f memberReplace) Available(config e2e.EtcdProcessClusterConfig, member e2e
...
+	getResp, err := cc.Get(ctx, "linearized-list-before-member-list")
+	if err != nil {
+		return 0, false, err
+	}
 	resp, err := cc.MemberList(ctx)
 	if err != nil {
 		return 0, false, err
 	}
+	if getResp.Header.MemberId != resp.Header.MemberId {
+		return 0, false, fmt.Errorf("expected Get and MemberList to be sent to seme member, got: %d and %d", getResp.Header.MemberId, resp.Header.MemberId)
```
Comment (ahrtr) [inline tests/robustness/failpoint/cluster.go:154]:
```suggestion
		return 0, false, fmt.Errorf("expected Get and MemberList to be sent to the same member, got: %d and %d", getResp.Header.MemberId, resp.Header.MemberId)
```

**Review by ahrtr — APPROVED** [review]
LGTM with a minor comment.

File: tests/robustness/failpoint/cluster.go (hunk 1 above)
Comment (serathius) [reply tests/robustness/failpoint/cluster.go]: Done

**Comment by serathius** [comment]
Ignoring arm64 failures as there are issues with infrastructure.

**Pull request merged**